spylib/oauth/router.py

```diff
@@ -115,29 +122,28 @@ async def shopify_callback(request: Request, args: Callback = Depends(Callback))
 
         # === If login ===
         # Get the online token from Shopify
-        online_token = await OnlineToken.get(
-            domain=args.shop,
-            code=args.code,
-            api_key=api_key,
-            api_secret_key=api_secret_key,
-        )
+        if OnlineToken:
+            online_token = await get_token(
+                domain=args.shop,
+                code=args.code,
+                api_key=api_key,
+                api_secret_key=api_secret_key,
+                CustomToken=OnlineToken,
+                Response=OnlineTokenResponse,
+            )
```

Comment:
Following the same approach as all our current other Shopify API calls, let's keep this function simple and have it just return JSON. It's role is to abstract the API call, not check the JSON itself, just getting it.
The Response class thing is a neat idea but it does the exact same validation as the OnlineToken, one line prior. That's overly complicated.
It seems like `get_token` doesn't even change if it's online or offline token ...

Now what you can do is catch the pydantic error if the JSON doesn't match what we want and add a message to clarify that!

 ```suggestion
             online_token_json = await get_token(
                 domain=args.shop,
                 code=args.code,
                 api_key=api_key,
                 api_secret_key=api_secret_key
             )
             online_token_json['store_name'] = domain_to_storename(args.shop)
             online_token = OnlineToken.parse_obj(online_token_json)
```